connection.go

```diff
@@ -144,6 +144,10 @@ func (mc *mysqlConn) interpolateParams(query string, args []driver.Value) (strin
 	buf = buf[:0]
 	argPos := 0
 
+	if len(args) != strings.Count(query, "?") {
+		return "", ErrArgsInvalid
```

Comment:
@methane  I'm sorry I don't quite catch you. The problem I find is that when the count of placehold(?) in sql is not equal to the count of args, there will be a index out of range panic.